Sources/AdventOfCode.swift

```diff
@@ -33,36 +36,46 @@ struct AdventOfCode: AsyncParsableCommand {
     allChallenges.max(by: { $0.day < $1.day })!
   }
 
-  func run(part: () async throws -> Any, named: String) async -> Duration {
-    var result: Result<Any, Error> = .success("<unsolved>")
+  func run<T>(part: () async throws -> T, named: String) async -> Duration {
+    var result: Result<T, Error>?
     let timing = await ContinuousClock().measure {
       do {
         result = .success(try await part())
       } catch {
         result = .failure(error)
       }
     }
-    switch result {
+    switch result! {
     case .success(let success):
       print("\(named): \(success)")
+    case .failure(let failure as PartUnimplemented):
+      print("Day \(failure.day) part \(failure.part) unimplemented")
     case .failure(let failure):
       print("\(named): Failed with error: \(failure)")
     }
     return timing
   }
 
   func run() async throws {
-    let challenge = try selectedChallenge
-    print("Executing Advent of Code challenge \(challenge.day)...")
+    let challenges =
+      if all {
+        allChallenges
+      } else {
+        try [selectedChallenge]
+      }
 
-    let timing1 = await run(part: challenge.part1, named: "Part 1")
-    let timing2 = await run(part: challenge.part2, named: "Part 2")
+    for challenge in challenges {
+      print("Executing Advent of Code challenge \(challenge.day)...")
 
-    if benchmark {
-      print("Part 1 took \(timing1), part 2 took \(timing2).")
-      #if DEBUG
-        print("Looks like you're benchmarking debug code. Try swift run -c release")
-      #endif
+      let timing1 = await run(part: challenge.part1, named: "Part 1")
+      let timing2 = await run(part: challenge.part2, named: "Part 2")
+
+      if benchmark {
+        print("Part 1 took \(timing1), part 2 took \(timing2).")
```

Comment:
We could move this print up into `func run`, the output might be better; Also I think the `#if DEBUG` print should only occur once as opposed to N days times.

Reply:
will merge this and do a follow-on

Reply:
ugh it's a pain without changing the `allChallenges` array to be of meta types or something